zerver/webhooks/patreon/view.py

```diff
@@ -0,0 +1,162 @@
+    )
+
+    return body
+
+
+# called when a patron deleted their membership
+def members_delete(payload: Dict[str, Any]) -> str:
+
+    # array of dicts
+    # first item contains the patron count
+    included = payload["included"][0]
+
+    # get the patron count
+    patron_count = included["attributes"]["patron_count"]
+
+    # create the body
+    body = (
+        f"A patron just deleted their membership. :notifications:\nPatrons in total: {patron_count}"
+    )
+
+    return body
+
+
+# called when a new patron joined through custom pledge
+def members_pledge_create(payload: Dict[str, Any]) -> str:
+
+    # array of dicts
+    # first item contains the patron count
+    included = payload["included"][0]
+
+    # get the patron count
+    patron_count = included["attributes"]["patron_count"]
+
+    # create the body
+    body = f"New patron joined through custom pledge! :tada:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+# called when a patron updated pledge
+def members_pledge_update(payload: Dict[str, Any]) -> str:
+
+    # array of dicts
+    # first item contains the patron count
+    included = payload["included"][0]
+
+    # get the patron count
+    patron_count = included["attributes"]["patron_count"]
+
+    # create the body
+    body = f"A patron just updated their pledge. :notifications:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+# called when a patron deleted pledge
+def members_pledge_delete(payload: Dict[str, Any]) -> str:
+
+    # array of dicts
+    # first item contains the patron count
+    included = payload["included"][0]
+
+    # get the patron count
+    patron_count = included["attributes"]["patron_count"]
+
+    # create the body
+    body = f"A patron just deleted their pledge. :notifications:\nPatrons in total: {patron_count}"
+
+    return body
+
+
+@webhook_view("Patreon")
+@has_request_variables
+def api_patreon_webhook(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    payload: Dict[str, Any] = REQ(argument_type="body"),
+) -> HttpResponse:
+
+    # get the http header
+    event = validate_extract_webhook_http_header(request, "X_PATREON_EVENT", "Patreon")
+    assert event is not None
+
+    # map the event to necessary function
+    handler = EVENTS_FUNCTION_MAPPER.get(event)
+    if handler is None:
+        raise UnsupportedWebhookEventType(event)
+
+    body = handler(payload)
+    topic = "notifications"
+
+    if body is not None:
+        check_send_webhook_message(request, user_profile, topic, body)
+
+    return json_success()
+
+
+EVENTS_FUNCTION_MAPPER: Dict[str, Callable[[Dict[str, Any]], Optional[str]]] = {
+    "members:create": members_create,
+    "members:update": members_update,
+    "members:delete": members_delete,
+    "members:pledge:create": members_pledge_create,
+    "members:pledge:update": members_pledge_update,
+    "members:pledge:delete": members_pledge_delete,
+}
```

Comment:
Um, this is just a small nitpick from me, but could you please move this to above the `api_patreon_webhook` function definition? In most other integrations this type of a map would be found either just above the `api_x_webhook` function or at the top of the file. Not really a hard-and-fast convention or anything, but it's just a thing that I (and I'm sure others) are sort of used to.